app/controllers/course/survey/responses_controller.rb

```diff
@@ -28,7 +28,12 @@ def edit
   end
 
   def update
-    @response.submit if params[:response][:submit]
+    if params[:response][:submit]
+      @response.submit
+    elsif params[:response][:unsubmit]
+      @response.unsubmit
```

Comment:
Small concern: since `unsubmit` is more of a single action - however, it is actually possible for a user to pass in updated response params, which seems as though it would get updated here as well. Even if the react components don't pass these params into the controller, it might still be cleaner to ensure the controller doesn't do this.